client/app/types/course/announcements.ts

```diff
@@ -0,0 +1,23 @@
+import { Permissions } from 'types';
+
+export type AnnouncementDataPermissions = Permissions<'canEdit' | 'canDelete'>;
+
+export interface AnnouncementData {
```

Comment:
Not AnnouncementListData?

Not sure why AnnouncementMiniEntity is not defined as well